### PR TITLE
fix(cdk): make `AbstractTuiControl` constructor DI compatible

### PR DESCRIPTION
**Pull request by waterplea**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes # 

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/addon-commerce/components/money/pipes/integer-part.pipe.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export class TuiIntegerPartPipe implements PipeTransform {
     ) {}
 
     transform(value: number): string {
-        return tuiFormatNumber(Math.floor(Math.abs(Number(value))), {
```
Comment (vladimirpotekhin) [inline projects/addon-commerce/components/money/pipes/integer-part.pipe.ts:15]: `Math.trunc(-15)` --> `-15`

File: projects/core/utils/format/format-number.ts [hunk 2]
```diff
@@ -20,7 +20,7 @@ export function tuiFormatNumber(
         ...TUI_DEFAULT_NUMBER_FORMAT,
         ...settings,
     };
-    const integerPartString = String(Math.floor(Math.abs(value)));
```
Comment (vladimirpotekhin) [inline projects/core/utils/format/format-number.ts:23]: same, seems like it'll add extra sign

File: projects/addon-commerce/components/money/pipes/integer-part.pipe.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ export class TuiIntegerPartPipe implements PipeTransform {
...
+        return tuiFormatNumber(Math.trunc(Number(value)), {
```
Comment (splincode) [inline projects/addon-commerce/components/money/pipes/integer-part.pipe.ts:15]: Please, could you use another pull request to have semantic changes in the codebase?

File: projects/addon-commerce/components/money/pipes/integer-part.pipe.ts (hunk 1 above)
Comment (waterplea) [reply projects/addon-commerce/components/money/pipes/integer-part.pipe.ts:15]: Oops

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**